### PR TITLE
WRKLDS-593: pkg/cli/admin/mustgather: Include must-gather namespace in inspect fallback

### PR DESCRIPTION
**Pull request by wking**
Carrying and adjusting #1273.

In [a CI run][1], [the must-gather failed][2]:

```
  [must-gather      ] OUT namespace/openshift-must-gather-9dzzn created
  [must-gather      ] OUT clusterrolebinding.rbac.authorization.k8s.io/must-gather-9kdhs created
  [must-gather      ] OUT pod for plug-in image registry.build01.ci.openshift.org/ci-op-4ryqvzt5/stable@sha256:c68fcf3161719f600a803627bca6d0ebf7894bf75282937aebe908a9ef54dedb created
  [must-gather-lxz2x] POD 2022-10-21T10:17:43.213934752Z Gathering data for ns/openshift-cluster-version...
  ...
  [must-gather-lxz2x] POD 2022-10-21T10:18:07.577813555Z Gathering data for ns/openshift-monitoring...
  [must-gather-lxz2x] OUT gather logs unavailable: read tcp 10.131.188.56:50182->3.12.171.103:6443: read: connection reset by peer
  [must-gather-lxz2x] OUT waiting for gather to complete
  [must-gather-lxz2x] OUT gather never finished: pods "must-gather-lxz2x" not found
  [must-gather      ] OUT namespace/openshift-must-gather-9dzzn deleted
  [must-gather      ] OUT clusterrolebinding.rbac.authorization.k8s.io/must-gather-9kdhs deleted
  Gathering data for ns/openshift-config...
```

And from [the build log][3]:

```
  Running must-gather...

  Error running must-gather collection:
      gather never finished for pod must-gather-lxz2x: pods "must-gather-lxz2x" not found

  Falling back to `oc adm inspect clusteroperators.v1.config.openshift.io` to collect basic cluster information.
```

This commit shifts the inspection fallback from 971dcc6020 (#749) to come before that temporary namespace is deleted, and to include the temporary namespace, so we capture events and such to help debug why the pod failed to launch or if/when something else deleted it.

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-ingress-operator/836/pull-ci-openshift-cluster-ingress-operator-master-e2e-aws-operator/1583384716713660416
[2]: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-ingress-operator/836/pull-ci-openshift-cluster-ingress-operator-master-e2e-aws-operator/1583384716713660416/artifacts/e2e-aws-operator/gather-must-gather/artifacts/must-gather/must-gather.log
[3]: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-ingress-operator/836/pull-ci-openshift-cluster-ingress-operator-master-e2e-aws-operator/1583384716713660416/artifacts/e2e-aws-operator/gather-must-gather/build-log.txt

### CODE REVIEW

File: pkg/cli/admin/mustgather/mustgather.go [hunk 1]
```diff
@@ -820,16 +833,24 @@ func (o *MustGatherOptions) newPod(node, image string) *corev1.Pod {
 
 // BackupGathering is called if the full must-gather has an error.  This is useful for making sure we get *something*
 // no matter what has failed.  It should be focused on universal openshift failures.
-func (o *MustGatherOptions) BackupGathering(ctx context.Context, errs []error) {
+func (o *MustGatherOptions) BackupGathering(ctx context.Context, namespace string, errs []error) {
+	targets := []string{
+		"clusterversion.v1.config.openshift.io",
```
Comment (wking) [inline pkg/cli/admin/mustgather/mustgather.go:838]: I'm also adding ClusterVersion to the fallback inspection, because it's often useful to know what the cluster-version operator is up to.  But I'm happy to spin that out into a separate pull request  if folks would prefer distinct pivots.

Comment (ardaguclu) [reply pkg/cli/admin/mustgather/mustgather.go:838]: It would be better to do that in separate PR.

**Comment by ardaguclu** [comment]
Thanks for this patch @wking. Has your change any impact on when `--run-namespace` is used?

File: pkg/cli/admin/mustgather/mustgather.go (hunk 1 above)
Comment (wking) [reply pkg/cli/admin/mustgather/mustgather.go:838]: I've spun this out into #2025.
